planner/core/planbuilder.go

```diff
@@ -513,6 +513,12 @@ type handleColHelper struct {
 	stackTail         int
 }
 
+func (hch *handleColHelper) resetForReuse() {
+	*hch = handleColHelper{
+		id2HandleMapStack: hch.id2HandleMapStack[:0],
+	}
+}
```

Comment:
```
hch.id2HandleMapStack = hch.id2HandleMapStack[:0]
hch.stackTail = 0
```
Could we reset `hch.id2HandleMapStack` directly?

Reply:
I do this deliberately.

```
*hch = handleColHelper{
		id2HandleMapStack: hch.id2HandleMapStack[:0],
}
```


When someone update the `handleColHelper` struct and add some more fields,
he may not notice the `resetForReuse` function, and forget to update it. 
Then things goes wrong.

While in my coding style, I mean reuse the `hch.id2HandleMapStack` field and only that field, reset other fields to the default value.

For example, when field `xxx string` is added, 

```
type handleColHelper struct {
    id2HandleMapStack []map[int64][]HandleCols
    stackTail int
    xxx string
}
```

Forget to update `resetForReuse` is not a serious mistake... the **code is robust**!